src/modding/ModConfigItemInfo.cs

```diff
@@ -0,0 +1,214 @@
+                case "colour":
+                case "c":
+                case "alphacolor":
+                case "alphacolour":
+                case "ac":
+                    var colorUIElement = childUIElement as ColorPickerButton;
+                    if (colorUIElement != null)
+                    {
+                        Value = colorUIElement.Color.ToHtml();
+                    }
+
+                    break;
+            }
+        }
+
+        return Value!;
+    }
+
+    /// <summary>
+    ///   Updates the UI element based on the Value variable
+    /// </summary>
+    public void UpdateUI()
+    {
+        if (GetChildCount() < 2)
+        {
+            return;
+        }
+
+        var childUIElement = GetChild(1);
+        switch (Type?.ToLower(CultureInfo.CurrentCulture) ?? string.Empty)
+        {
+            case "int":
+            case "integer":
+            case "i":
+            case "int range":
+            case "integer range":
+            case "ir":
+            case "float":
+            case "f":
+            case "float range":
+            case "fr":
+                var numberUIElement = childUIElement as Range;
+                if (numberUIElement != null)
+                {
+                    numberUIElement.Value = Convert.ToDouble(Value, CultureInfo.CurrentCulture);
+                }
+
+                break;
+            case "bool":
+            case "boolean":
+            case "b":
+                var buttonUIElement = childUIElement as Button;
+                if (buttonUIElement != null)
+                {
+                    buttonUIElement.Pressed = Convert.ToBoolean(Value, CultureInfo.CurrentCulture);
+                }
+
+                break;
+            case "string":
+            case "s":
+                var stringUIElement = childUIElement as LineEdit;
+                if (stringUIElement != null)
+                {
+                    stringUIElement.Text = Convert.ToString(Value, CultureInfo.CurrentCulture);
+                }
+
+                break;
+            case "option":
+            case "enum":
+            case "o":
+                var optionUIElement = childUIElement as OptionButton;
+                if (optionUIElement != null)
+                {
+                    optionUIElement.Selected = Convert.ToInt32(Value, CultureInfo.CurrentCulture);
+                }
+
+                break;
+            case "color":
+            case "colour":
+            case "c":
+            case "alphacolor":
+            case "alphacolour":
+            case "ac":
+                var colorUIElement = childUIElement as ColorPickerButton;
+                if (colorUIElement != null && Value != null)
+                {
+                    colorUIElement.Color = new Color((string)Value);
+                }
+
+                break;
+        }
+    }
+
+    /// <summary>
+    ///   Returns a list of string based on the options the OptionButton can have
```

Comment:
```suggestion
    ///   Returns a list of strings based on the options the OptionButton can have
```

Also, I'm not sure this helps me in understanding what this method does. Does it returns all options of the `OptionButton` as a String?

Reply:
Yes, how do you think I can make it more understandable

Reply:
The above was my way to clarify it. Perhaps say it returns each option of the OptionButton as a string, and precise whatever this string would be? In the end, you are the one who knows best how to explain your code. ;)